python/tests/io/test_pyarrow.py

```diff
@@ -16,16 +16,39 @@
 # under the License.
 # pylint: disable=protected-access,unused-argument,redefined-outer-name
 
+import math
 import os
 import tempfile
-from typing import Any, List, Optional
+import uuid
+from datetime import (
```

Comment:
@Fokko, why are there `datetime` classes here?

I think that tests like these should completely avoid Python datetime logic and instead should directly use the internal representations.

Reply:
We use this for PyArrow which has the datetime classes in their public API.